rcmd/install.go

```diff
@@ -534,44 +534,123 @@ func writeDescriptionInfo(ir InstallRequest, ia InstallArgs) {
+
+	fileObj, err := fs.OpenFile(filename, os.O_WRONLY,0755)
+	defer fileObj.Close()
+
+	if err != nil {
+		log.WithFields(log.Fields{
+			"filename" : filename,
+		}).Error("could not update DESCRIPTION file", err)
+		return err
+	}
+	writer := bufio.NewWriter(fileObj)
+
+	newlineChar := getEOLcharacter()
+	for _, line := range update {
+		lineWithEOL := line + newlineChar
+		_, err := writer.WriteString(lineWithEOL)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"filename" : filename,
+				"line" : lineWithEOL,
+			}).Error("problem inserting line into DESCRIPTION file", err)
+			return err
 		}
-		update = append(update, []byte("PkgrVersion: "+version+"\n")...)
-		update = append(update, []byte("PkgrInstallType: "+installType+"\n")...)
-		update = append(update, []byte("PkgrRepositoryURL: "+repoURL+"\n")...)
+	}
+	err = writer.Flush()
+	if err != nil {
+		log.WithFields(log.Fields{
+			"filename" : filename,
+		}).Error("could not update DESCRIPTION file", err)
+		return err
+	}
 
-		if writeFile {
-			fi, _ := os.Stat(filename)
-			err = afero.WriteFile(fs, filename, update, fi.Mode())
-			if err != nil {
-				log.Warn(fmt.Sprintf("Error updating DESCRIPTION file:%s", err))
-			}
+	return nil
+	//
+	//fi, _ := os.Stat(filename)
+	//err := afero.WriteFile(fs, filename, update, fi.Mode())
+	//if err != nil {
+	//	log.Warn(fmt.Sprintf("Error updating DESCRIPTION file:%s", err))
+	//}
+	//return err
+}
+
+// takes a string slice of lines (without line-endings) representing a DESCRIPTION file and uses
+// provided parameters to make appropriate updates to that description file.
+// returns the updated string slice, still without new-line characters.
+func updateDescriptionInfoByLines(lines []string, version, installType, repoURL, repo string) []string {
+
+	var newLines []string
+	for _, line := range lines {
+		//log.Info("Starting repo: " + repo)
+
+		// Don't add these lines to final set, if they exist. This way, we can add/"overwrite" them at the end.
+		if strings.Contains(line, "PkgrVersion:") ||
+			strings.Contains(line, "PkgrInstallType:") ||
+			strings.Contains(line, "PkgrRepositoryURL") {
+			continue
 		}
+
+		if strings.Contains(line, string("Repository:")) && !strings.Contains(line, repo) {
+			//log.Info("Got where we needed.")
+			originalRepo := strings.Trim(strings.Split(line, ":")[1], " ")
+			newLines = append(newLines, "OriginalRepository: " + originalRepo)
+			line = "Repository: " + repo
+		}
+		newLines = append(newLines, line)
 	}
-	return update, err
+
+	newLines = append(newLines, "PkgrVersion: "+version)
+	newLines = append(newLines, "PkgrInstallType: "+installType)
+	newLines = append(newLines, "PkgrRepositoryURL: "+repoURL)
+
+	return newLines
 }
+
+// function to inspect the current operating system and attempt to determine appropriate line-endings.
+// if Windows, returns "\r\n". Otherwise, returns "\n".
+func getEOLcharacter() string {
```

Comment:
@dpastoor A few things here:
1) Newlines are trimmed by Scanner (including carriage returns, if they're present), so simply trying to modify the lines in-place doesn't work unless we're scanning in as bytes.
2) Additionally, we have to add a new line for OriginalRepository no matter how you look at it, so we need a way to access the correct newline character for the OS. fmt.Fprint() and such uses \n behind the scenes (I checked), so that wasn't an option. 
3) Since newlines are a problem elsewhere in the code, I don't expect this to be the final resting place for this function, but I'm not sure where it ought to go to avoid circular dependencies, and I didn't want to commit to making a new top-level folder. So basically, we should discuss this.

Reply:
Irrelevent -- we are comfortable always writing out using \n, and we are using ReadLinesFS from goutils now to more robustly parse _in_ the files line-by-line.